packages/toolpad-app/src/components/Home.tsx

```diff
@@ -112,45 +120,178 @@ function AppDeleteDialog({ app, onClose }: AppDeleteDialogProps) {
+        <CardHeader
+          action={
+            <IconButton
+              aria-label="settings"
+              aria-controls={menuOpen ? 'basic-menu' : undefined}
+              aria-haspopup="true"
+              aria-expanded={menuOpen ? 'true' : undefined}
+              onClick={handleMenuClick}
+            >
+              <MoreVertIcon />
+            </IconButton>
+          }
+          disableTypography
+          title={
+            editingTitle ? (
+              <TextField
+                variant="standard"
+                size="small"
+                inputRef={appTitleInput}
+                sx={{ paddingBottom: '4px' }}
+                InputProps={{ sx: { fontSize: '1.5rem', height: '1.5em' } }}
+                onKeyUp={handleAppTitleInput}
+                onBlur={handleAppTitleBlur}
+                defaultValue={appTitle}
+              />
+            ) : (
+              <Typography gutterBottom variant="h5" component="div">
+                {app ? appTitle : <Skeleton />}
+              </Typography>
+            )
+          }
+          subheader={
+            <Typography variant="body2" color="text.secondary">
+              {app ? `Created: ${app.createdAt.toLocaleDateString('short')}` : <Skeleton />}
+            </Typography>
+          }
+        />
+
+        <CardActions>
+          <Button
+            size="small"
+            component="a"
+            href={app ? `/_toolpad/app/${app.id}/editor` : ''}
+            disabled={!app}
+          >
+            Edit
+          </Button>
+          <Button size="small" component="a" href={app ? `deploy/${app.id}` : ''} disabled={!app}>
+            View
+          </Button>
+        </CardActions>
+      </Card>
+      <Menu
+        id="basic-menu"
+        anchorEl={menuAnchorEl}
+        open={menuOpen}
+        onClose={handleMenuClose}
+        MenuListProps={{
+          'aria-labelledby': 'basic-button',
+          dense: true,
+        }}
+      >
+        <MenuItem onClick={handleRenameClick}>
+          <DriveFileRenameOutlineIcon sx={AppCardMenuIconStyles} />
+          Rename
+        </MenuItem>
+        <MenuItem onClick={handleMenuClose}>
+          <FileCopyIcon sx={AppCardMenuIconStyles} /> Duplicate
+        </MenuItem>
+        <MenuItem onClick={handleDeleteClick}>
+          <DeleteIcon sx={AppCardMenuIconStyles} />
+          Delete
+        </MenuItem>
+      </Menu>
+    </React.Fragment>
   );
 }
 
 export default function Home() {
-  const { data: apps = [], status, error } = client.useQuery('getApps', []);
+  const [apps, setApps] = React.useState<App[]>([]);
+
+  const { status, error } = client.useQuery('getApps', [], {
+    onSuccess: (data: App[]) => setApps(data),
+  });
 
   const [createDialogOpen, setCreateDialogOpen] = React.useState(false);
 
   const [deletedApp, setDeletedApp] = React.useState<null | App>(null);
 
+  const onUpdate = React.useCallback((updatedApp: App) => {
+    setApps((previousApps) => {
```

Comment:
You should refetch the app list instead, so that the cache gets renewed. Like in https://github.com/mui/mui-toolpad/blob/3ebb35665c46646ace00b2c972cd1f0d637c4f6b/packages/toolpad-app/src/components/Home.tsx#L91

You can even directly use the `refetch` method returned by `client.useQuery('getApps'` in this case.